### ISSUE TITLE
Add SystemTest to run build-namelist_test.pl

### ISSUE THREAD

**Issue by samsrabin** [ESCOMP/CTSM#2825]
It's easy to forget to run `build-namelist_test.pl` before merging. It would be nice if we had a SystemTest to do this so that we could add a test in `aux_clm` to run it for us, as we do for the FORTRAN unit tests (`FUNITCTSM`).

**Comment by ekluzek** [comment]
Nice I love this as an idea. It might not be too hard to do. And actually might save time. Personally I end up doing a bunch of work to save baselines for it for previous versions. This would already be handed if it was in a SystemTest.

As a Github workflow it could probably be done to just run the list without doing the baseline compare part.

**Comment by samsrabin** [comment]
Yeah, I was looking at `FUNITCTSM` and the `FUNIT` class it inherits, and it doesn't seem too bad at all.
